cpp/open3d/t/io/file_format/FileSPLAT.cpp

```diff
@@ -0,0 +1,354 @@
+// ----------------------------------------------------------------------------
+// -                        Open3D: www.open3d.org                            -
+// ----------------------------------------------------------------------------
+// Copyright (c) 2018-2024 www.open3d.org
+// SPDX-License-Identifier: MIT
+// ----------------------------------------------------------------------------
+
+#include <filesystem>
+#include <iostream>
+#include <vector>
+
+#include "open3d/core/Dtype.h"
+#include "open3d/core/Tensor.h"
+#include "open3d/io/FileFormatIO.h"
+#include "open3d/t/geometry/TensorMap.h"
+#include "open3d/t/io/PointCloudIO.h"
+#include "open3d/utility/FileSystem.h"
+#include "open3d/utility/Logging.h"
+#include "open3d/utility/ProgressReporters.h"
+
+namespace open3d {
+namespace t {
+namespace io {
+
+#define SH_C0 0.28209479177387814
+#define SPLAT_GAUSSIAN_BYTE_SIZE 32
+
+struct AttributePtr {
+    AttributePtr(const core::Dtype &dtype,
+                 const void *data_ptr,
+                 const int &group_size)
+        : dtype_(dtype), data_ptr_(data_ptr), group_size_(group_size) {}
+
+    const core::Dtype dtype_;
+    const void *data_ptr_;
+    const int group_size_;
+};
+
+// Sigmoid function for opacity calculation
+inline double sigmoid(double x) { return 1.0 / (1.0 + std::exp(-x)); }
+
+// Function to convert int to byte representation
+unsigned char int_to_byte(const int val) {
+    unsigned char byte_val = 0;
+
+    if (val > std::numeric_limits<unsigned char>::min() &&
+        val < std::numeric_limits<unsigned char>::max()) {
+        byte_val = static_cast<unsigned char>(val);
+    }
+
+    return byte_val;
+}
+
+template <typename scalar_t>
+Eigen::Vector4i ComputeColor(const scalar_t *f_dc_ptr,
+                             const scalar_t *opacity_ptr) {
+    Eigen::Vector4f color;
+
+    color[0] = 0.5 + SH_C0 * f_dc_ptr[0];
+    color[1] = 0.5 + SH_C0 * f_dc_ptr[1];
+    color[2] = 0.5 + SH_C0 * f_dc_ptr[2];
+    color[3] = sigmoid(*opacity_ptr);
+
+    // Convert color to uint8 (scale, clip, and cast)
+    return (color * 255.0).cwiseMin(255.0).cwiseMax(0.0).cast<int>();
+}
+
```

Comment:
Use C++ fstreams with exceptions for error handling for faster IO.

Reply:
Changed it to use fstreams with exception handling.